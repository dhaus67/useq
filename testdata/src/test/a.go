package test

import (
	"fmt"
	"os"
)

func test() {
	name := "John"

	fmt.Printf("This is my name: \"%s\"\n", name) // want `use %q instead of \"%s\" for formatting strings with quotations`
	fmt.Printf("This is my name: %s", name)
	fmt.Printf("This is my name: %q", name)
	fmt.Printf("This is my age: %d", 42)

	fmt.Sprintf("This is my name: \"%s\"\n", name) // want `use %q instead of \"%s\" for formatting strings with quotations`
	fmt.Sprintf("This is my name: %s", name)
	fmt.Sprintf("This is my name: %q", name)
	fmt.Sprintf("This is my age: %d", 42)

	fmt.Fprintf(os.Stdout, "This is my name: \"%s\"\n", name) // want `use %q instead of \"%s\" for formatting strings with quotations`
	fmt.Fprintf(os.Stdout, "This is my name: %s", name)
	fmt.Fprintf(os.Stdout, "This is my name: %q", name)
	fmt.Fprintf(os.Stdout, "This is my age: %d", 42)

	fmt.Errorf("This is my name: \"%s\"\n", name) // want `use %q instead of \"%s\" for formatting strings with quotations`
	fmt.Errorf("This is my name: %s", name)
	fmt.Errorf("This is my name: %q", name)
	fmt.Errorf("This is my age: %d", 42)
}
