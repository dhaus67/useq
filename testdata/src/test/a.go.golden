// Copyright 2024 dhaus67
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

package test

import (
	"fmt"
	"os"
)

func myCustomPrint(b bool, msg string, format ...any) {
	if b {
		fmt.Printf(msg, format...)
	}
}

func test() {
	name := "John"

	fmt.Printf("This is my name: %q\n", name) // want `use %q instead of \"%s\" for formatting strings with quotations`
	fmt.Printf("This is my name: %s", name)
	fmt.Printf("This is my name: %q", name)
	fmt.Printf("This is my age: %d", 42)

	fmt.Sprintf("This is my name: %q\n", name) // want `use %q instead of \"%s\" for formatting strings with quotations`
	fmt.Sprintf("This is my name: %s", name)
	fmt.Sprintf("This is my name: %q", name)
	fmt.Sprintf("This is my age: %d", 42)

	fmt.Fprintf(os.Stdout, "This is my name: %q\n", name) // want `use %q instead of \"%s\" for formatting strings with quotations`
	fmt.Fprintf(os.Stdout, "This is my name: %s", name)
	fmt.Fprintf(os.Stdout, "This is my name: %q", name)
	fmt.Fprintf(os.Stdout, "This is my age: %d", 42)

	fmt.Errorf("This is my name: %q\n", name) // want `use %q instead of \"%s\" for formatting strings with quotations`
	fmt.Errorf("This is my name: %s", name)
	fmt.Errorf("This is my name: %q", name)
	fmt.Errorf("This is my age: %d", 42)

	myCustomPrint(true, "This is my name: %q\n", name) // want `use %q instead of \"%s\" for formatting strings with quotations`
	myCustomPrint(true, "This is my name: %s", name)
	myCustomPrint(true, "This is my name: %q", name)
	myCustomPrint(true, "This is my age: %d", 42)
}
